src/AlgebraicGeometry/Schemes/AffineSchemeOpenSubscheme/Rings/Methods.jl

```diff
@@ -132,6 +138,14 @@ function ==(a::T, b::T) where {T<:AffineSchemeOpenSubschemeRingElem}
   return true
 end
 
+function Base.hash(a::AffineSchemeOpenSubschemeRingElem, h::UInt)
+  h = hash(parent(a), h)
+  for f in restrictions(a)
+    h = hash(f, h)
+  end
```

Comment:
```suggestion
```
I'm afraid I do not fully understand the last comment. I would just remove this block here. Then I think everything is good to go.